unit-tests/syncer/test-ts-eof.py

```diff
@@ -0,0 +1,150 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2021 Intel Corporation. All Rights Reserved.
+
+import pyrealsense2 as rs
+from rspy import log, test
+import sw
+
+
+# The timestamp jumps are closely correlated to the FPS passed to the video streams:
+# syncer expects frames to arrive every 1000/FPS milliseconds!
+sw.fps_c = sw.fps_d = 60
+sw.init()
+
+import tempfile, os
+temp_dir = tempfile.TemporaryDirectory( prefix = 'recordings_' )
+filename = os.path.join( temp_dir.name, 'rec.bag' )
+recorder = rs.recorder( filename, sw.device )
+
+sw.start()
+
+#############################################################################################
+#
+test.start( "Init" )
+
+# It can take a few frames for the syncer to actually produce a matched frameset (it doesn't
+# know what to match to in the beginning)
+
+sw.generate_depth_and_color( frame_number = 0, timestamp = 0 )
+sw.expect( depth_frame = 0 )                          # syncer doesn't know about color yet
+sw.expect( color_frame = 0, nothing_else = True )     # less than next expected of D
+#
+# NOTE: if the syncer queue wasn't 100 (see above) then we'd only get the color frame!
+#
+sw.generate_depth_and_color( 1, sw.gap_d * 1 )
+sw.expect( depth_frame = 1, color_frame = 1, nothing_else = True )  # frameset 1
+
+test.finish()
+#
+#############################################################################################
+#
+test.start( "Keep going" )
+
+sw.generate_depth_and_color( 2, sw.gap_d * 2 )
+sw.expect( depth_frame = 2, color_frame = 2, nothing_else = True )  # frameset 2
+
+test.finish()
+#
+#############################################################################################
+#
+test.start( "Stop giving color; nothing output" )
+
+sw.generate_depth_frame( 3, sw.gap_d * 3 )
+
+# The depth frame will be kept in the syncer, and never make it out (no matching color frame
+# and we're not going to push additional frames that would cause it to eventually flush):
+#
+sw.expect_nothing()
+#
+# ... BUT the file should still have it!
+
+test.finish()
+#
+#############################################################################################
+#
+test.start( "Dump the file" )
+
+recorder.pause()
+recorder = None  # otherwise the file will be open when we exit
+log.d( "filename=", filename )
+sw.stop()
+sw.reset()
+#
+# Dump it... should look like:
+#     [Depth/0 #0 @0.000000]
+#     [Color/1 #0 @0.000000]
+#     [Depth/0 #1 @16.666667]
+#     [Color/1 #1 @16.666667]
+#     [Depth/0 #2 @33.333333]
+#     [Color/1 #2 @33.333333]
+#     [Depth/0 #3 @50.000000]    <--- the frame that was "lost"
```

Comment:
Who is printing this lines? The syncer?  I don't see a print on `reset` function